core/envelope-sim/src/test/java/fr/sncf/osrd/envelope/EnvelopePartTest.java

```diff
@@ -109,4 +114,43 @@ void testGetMechanicalEnergyConsumed() {
             assertEquals(expectedEnvelopePartEnergy, envelopePartEnergy, 0.1 * expectedEnvelopePartEnergy + 1000);
         }
     }
+
+    @ParameterizedTest
+    @MethodSource("minEnvelopePartsArgs")
+    void testMinEnvelopeParts(
+            EnvelopePart envelopePartA,
+            EnvelopePart envelopePartB,
+            Stream<Integer> intersectionIndexes,
+            double[] expectedPositions,
+            double[] expectedSpeeds) {
+        var minEnvelope = EnvelopePart.min(
+                envelopePartA, envelopePartB, Collections.singleton(envelopePartA.getAttr(EnvelopeProfile.class)));
+        var resultingPositions = minEnvelope.clonePositions();
+        var resultingSpeeds = minEnvelope.cloneSpeeds();
+
+        intersectionIndexes.forEach(intersectionIndex -> {
+            var intersectionPosition = resultingPositions[intersectionIndex];
+            var intersectionSpeed = resultingSpeeds[intersectionIndex];
+            assertEquals(envelopePartA.interpolateSpeed(intersectionPosition), intersectionSpeed);
+            assertEquals(envelopePartB.interpolateSpeed(intersectionPosition), intersectionSpeed);
+        });
+        assertThat(expectedPositions).isEqualTo(resultingPositions);
+        assertThat(expectedSpeeds).isEqualTo(resultingSpeeds);
+    }
+
+    static Stream<Arguments> minEnvelopePartsArgs() {
+        return Stream.of(
+                Arguments.of(
+                        EnvelopeTestUtils.generateTimes(new double[] {0, 10, 20}, new double[] {100, 50, 0}),
+                        EnvelopeTestUtils.generateTimes(new double[] {5, 14, 16}, new double[] {150, 50, 0}),
+                        Stream.of(4),
+                        new double[] {0, 5, 10, 14, 15, 16},
+                        new double[] {100, 79.05694150420949, 50, 38.72983346207417, 35.35533905932738, 0}),
+                Arguments.of(
+                        EnvelopeTestUtils.generateTimes(new double[] {5, 14, 16}, new double[] {100, 50, 0}),
+                        EnvelopeTestUtils.generateTimes(new double[] {0, 10, 20}, new double[] {150, 50, 0}),
+                        Stream.of(1, 4),
+                        new double[] {5, 7.142857142857143, 10, 14, 15, 16},
+                        new double[] {100, 90.63269671749657, 50, 38.72983346207417, 35.35533905932738, 0}));
```

Comment:
Unexpected output here too, if using `min(beginPositions)` in case of equality (both speeds are 100 at position 5):
```suggestion
                        new double[] {0, 5, 10, 14, 15, 16},
                        new double[] {150, 100, 50, 30, 25, 0}));
```

Reply:
They aren't. See comment above (yes it's kinda cursed but w/e). Trust the assertions at intersection.

EDIT: ~~o.o this doesn't look good~~ nvm it was correct as expected.